packages/client/src/InfluxDBClient.ts

```diff
@@ -0,0 +1,134 @@
+      this._options = {url: options}
+    } else if (options !== null && typeof options === 'object') {
+      this._options = options
+    } else {
+      throw new IllegalArgumentError('No url or configuration specified!')
+    }
+    const url = this._options.url
+    if (typeof url !== 'string')
+      throw new IllegalArgumentError('No url specified!')
+    if (url.endsWith('/')) this._options.url = url.substring(0, url.length - 1)
+    this._queryApi = new QueryApiImpl(this._options)
+    this.transport = this._options.transport ?? new TransportImpl(this._options)
+    this._writeApi = new WriteApiImpl(this.transport)
+  }
+
+  private _mergeWriteOptions = (writeOptions?: Partial<WriteOptions>) => {
+    return {
+      ...this._options.writeOptions,
+      ...writeOptions,
+    }
+  }
+
+  async write(
+    database: string,
+    data: // The 'ArrayLike' and 'PointRecord' types must be marked with 'Not' types to avoid type confusion
+    | NotPointRecord<
+          ArrayLike<string> | ArrayLike<Point> | ArrayLike<PointRecord>
+        >
+      | NotArrayLike<PointRecord>
+      | string
+      | Point,
+    org?: string,
+    writeOptions?: Partial<WriteOptions>
+  ): Promise<void> {
+    const arrayData: ArrayLike<any> = (
+      isArrayLike(data) && typeof data !== 'string' ? data : [data]
+    ) as any[]
+    if (arrayData.length === 0) return
+    if (typeof arrayData[0] === 'string') {
+      await this.writeLines(database, arrayData as any, org, writeOptions)
+    } else if (arrayData[0] instanceof Point) {
+      await this.writePoints(database, arrayData as any, org, writeOptions)
+    } else {
+      await this.writeRecords(database, arrayData as any, org, writeOptions)
+    }
+  }
+
+  async writeLines(
+    database: string,
+    lines: string | ArrayLike<string>,
+    org?: string,
+    writeOptions?: Partial<WriteOptions>
+  ) {
+    await this._writeApi.doWrite(
+      database,
+      typeof lines === 'string' ? [lines] : Array.from(lines),
+      org,
+      this._mergeWriteOptions(writeOptions)
+    )
+  }
+
+  async writePoints(
+    database: string,
+    points: ArrayLike<Point> | Point,
+    org?: string,
+    writeOptions?: Partial<WriteOptions>
+  ): Promise<void> {
+    const pointsArr = points instanceof Point ? [points] : Array.from(points)
+
+    await this._writeApi.doWrite(
+      database,
+      pointsArr.map((p) => p.toLineProtocol()).filter(isDefined),
+      org,
+      this._mergeWriteOptions(writeOptions)
+    )
+  }
+
+  async writeRecords(
+    database: string,
+    records: ArrayLike<PointRecord>,
+    org?: string,
+    writeOptions?: Partial<WriteOptions>
+  ): Promise<void> {
+    await this.writePoints(
+      database,
+      Array.from(records).map((record) => Point.fromRecord(record)),
+      org,
+      writeOptions
+    )
+  }
+
+  query(
+    database: string,
+    query: string,
+    queryType: QueryType = 'sql'
+  ): AsyncGenerator<Map<string, any>, void, void> {
```

Comment:
Please change return type to array of objects. It will be more consistent with other clients:

 ```suggestion
  ): AsyncGenerator<[any], void, void> {
```

In the next version we will add support for "structured" object. For more info see: https://github.com/orgs/influxdata/projects/108/views/1?pane=issue&itemId=31487801